core-bundle/src/Resources/contao/pages/PageError404.php

```diff
@@ -76,14 +80,23 @@ public function getResponse()
 	/**
 	 * Prepare the output
 	 *
+	 * @param PageModel|null $page
+	 *
 	 * @return PageModel
 	 *
 	 * @internal Do not call this method in your code. It will be made private in Contao 5.0.
 	 */
-	protected function prepare()
+	protected function prepare($page=null)
 	{
-		// Find the matching root page
-		$objRootPage = $this->getRootPageFromUrl();
+		if ($page instanceof PageModel && $page->type === 'error_404')
```

Comment:
```suggestion
		$obj404 =  null;

		if ($page instanceof PageModel && $page->type === 'error_404')
```

Otherwise you will get a `variable $obj404 might not be defined` warning further down.